src/documentation/sdks/euphoria.md

```diff
@@ -18,72 +18,530 @@ See the License for the specific language governing permissions and
+`Dataset`. Lets look at simple `MapElements` example.
+
+```java
+Dataset<Integer> input = ...
+
+Dataset<String> mappedElements =
+  MapElements
+    .named("Int2Str")
+    .of(input)
+    .using(String::valueOf)
+    .output();
+```
+The operator consumes `input`, it applies given lambda expression (`String::valueOf`) on each element of `input` and returns mapped `Datatset`. Developer is guided through series of steps when creating operator so the declaration of an operator is straightforward. To start building operator just wrote its name and '.'. Your IDE will give you hits.
+
+First step to build any operator is to give it a name through `named()` method. This is optional but recommended step. The name is propagated through system and can latter be used when debugging.
+
+### Coders and Types
+Beam's Java SDK requires developers to supply `Coder` for custom element type in order to have a way of materializing elements. Euphoria integrates [Kryo](https://github.com/EsotericSoftware/kryo) as default way of serialization. Recommended way of using [Kryo](https://github.com/EsotericSoftware/kryo) is to register all the types which will Kryo serialize. Sometimes it is also a good idea to register Kryo serialisers of its own too. Euphoria allows you to do that by implementing your own `KryoRegistrar` and using it as input to `RegisterCoders`. The `RegisterCoders` can also be used as convenient way of supplying Beam `Coder` for specified element types.
+```java
+RegisterCoders
+  .to(flow)
+  .setKryoClassRegistrar(
+    (kryo) -> {
+      kryo.register(KryoSerializedElementType.class); //other may follow
+    })
+  .registerCoder(AnotherElementType.class, beamCoder)
+  .registerCoder(new TypeDescriptor<ParametrizedTestDataType<String>>() {}, typeParametrizedCoder)
+  .done();
+```
+When fast prototyping you may decide not to care much about coders and Euphoria will use Kryo for every element type which do not have registered `Coder`. Even for elements of types not registered with Kryo. That of course degrades performance, since Kryo is not able to serialize instances of unregistered types effectively. This behavior is enabled by default and can be disabled when creating `BeamFlow`.
+```java
+BeamFlow flow = BeamFlow.of(pipeline, false); // set `allowKryoCoderAsFallback` param to false
+```
+Euphoria resolves coder using types of elements. Type information is not available at runtime when element type is described by lambda implementation. It is due to type erasure and dynamic nature of lambda expressions. So there is an optional way of supplying `TypeDescriptor` every time new type is introduced during Operator construction.
+```java
+Dataset<Integer> input = ...
+
+MapElements
+  .named("Int2Str")
+  .of(input)
+  .using(String::valueOf, TypeDescriptors.strings())
+  .output();
+```
+Supplying `TypeDescriptors` will become mandatory when using Kryo as fallback `Coder` is disabled.
```

Comment:
Supplying `TypeDescriptors` becomes mandatory, when usage of Kryo

Reply:
Fixed.